src/business-logic/services/rundown-lock-service.ts

```diff
@@ -0,0 +1,125 @@
+import { Part } from '../../model/entities/part'
+import { Piece } from '../../model/entities/piece'
+import { Owner } from '../../model/enums/owner'
+import { InTransition } from '../../model/value-objects/in-transition'
+import { RundownService } from './interfaces/rundown-service'
+
+const RUNDOWN_LOCK_INTERVAL = 500
+const RUNDOWN_LOCK_ERROR_TEXT = 'Unable to do action. An action was already executed less than 500ms ago'
+
+export class RundownLockService implements RundownService {
+  private static instance: RundownService
+  private hasEnoughTimePassed: boolean = true
+
+  constructor(private readonly rundownService: RundownService) {}
+
+  public static getInstance(rundownService: RundownService): RundownService {
+    if (!this.instance) {
+      this.instance = new RundownLockService(rundownService)
+    }
+    return this.instance
+  }
```

Comment:
The order variables and static methods is not ordered after our conventions.
For singletons we order it in this order:
- private instance
- public getInstance
- instance variables
- constructor